### PR TITLE
[12.0][MIG] agreement_sale

### PR DESCRIPTION
**Pull request by bealdav**

### CODE REVIEW

File: agreement_sale/models/sale.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from odoo import models, fields
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    agreement_id = fields.Many2one(
```
Comment (murtuzasaleh) [inline agreement_sale/models/sale.py:11]: @bealdav  Can I remove [agreement_id](https://github.com/OCA/contract/pull/294/files#diff-c4b698254a164c50c4c83c3cb1965539R14) field from my module? (agreement_legal_sale)
If yes then can you please pass copy=False attribute on this field.
Also, This PR is ready for review or in WIP?

@max3903

File: agreement_sale/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    'name': 'Agreement Sale',
+    'summary': "Agreement on sales",
+    'version': '12.0.1.0.0',
+    'category': 'Contract',
+    'author': 'Akretion, '
+              'Odoo Community Association (OCA)',
+    'website': 'https://github.com/oca/contract',
+    'license': 'AGPL-3',
+    'depends': [
+        'sale',
```
Comment (murtuzasaleh) [inline agreement_sale/__manifest__.py]: @bealdav Please replace `sale_management` module instead of `sale` module.

**Review by murtuzasaleh — CHANGES_REQUESTED** [review]

**Comment by murtuzasaleh** [comment]
@bealdav Could you maybe change your PR title to be a little bit clearer? If it is migration and ready to review to 12, you can say, [12.0][MIG] agreement sale or If it is WIP [12.0][WIP][MIG] agreement sale.

File: agreement_sale/views/agreement.xml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+
+<odoo>
+
+    <menuitem id="agreement.agreement_menu" action="agreement.agreement_action"
```
Comment (murtuzasaleh) [inline agreement_sale/views/agreement.xml:5]: @max3903 Is it fine under the sale order configuration?

File: agreement_sale/models/sale.py (hunk 1 above)
Comment (bealdav) [reply agreement_sale/models/sale.py:11]: Done

**Comment by bealdav** [comment]
Ready for review. Thanks

File: agreement_sale/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    'license': 'AGPL-3',
+    'depends': [
+        'sale_management',
+        'agreement',
+    ],
+    'data': [
+        'security/ir.model.access.csv',
+        'views/agreement.xml',
+        ],
+    'demo': [],
```
Comment (nikul-serpentcs) [inline agreement_sale/__manifest__.py]: if not any file then no need `demo`

**Review by nikul-serpentcs — CHANGES_REQUESTED** [review]
netpicking

**Comment by bealdav** [comment]
Please while review this PR, use 'Insert a suggestion' then fix is easier

Thanks a lot

File: agreement_sale/views/agreement.xml (hunk 3 above)
Comment (max3903) [reply agreement_sale/views/agreement.xml:5]: Yes

File: agreement_sale/models/sale.py (hunk 1 above)
Comment (max3903) [reply agreement_sale/models/sale.py:11]: @murtuzasaleh Yes you can.

File: agreement_sale/__manifest__.py (hunk 4 above)
Comment (murtuzasaleh) [reply agreement_sale/__manifest__.py]: @bealdav LGTM, Please remove `demo`.

**Review by ygol — APPROVED** [review]

**Comment by ygol** [comment]
I approved it a bit too hasty :(
by changing the main agreement menu it impact agreement_legal's menu.
Maybe we should then remove the main menu from "agreement" and let sub-modules handle it? I would then add the "main" agreement menu in "agreement_legal"

File: agreement_sale/security/ir.model.access.csv [hunk 5]
```diff
@@ -0,0 +1,2 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+agreement.access_agreement_full,Full access on agreement grp,agreement.model_agreement,sales_team.group_sale_manager,1,1,1,1
```
Comment (max3903) [inline agreement_sale/security/ir.model.access.csv]: Please remove the newline.

**Comment by max3903** [comment]
@ygol I am not sure to follow you here. We can rearrange the menu as we want in `agreement_legal_sale`

**Review by osi-scampbell — APPROVED** [review]

**Comment by ygol** [comment]
> 
> 
> @ygol I am not sure to follow you here. We can rearrange the menu as we want in `agreement_legal_sale`

when you install agreement_sale and then also agreement_legal_sale (for example) because of the dependencies you end with no agreement main menu since agreement_sale moved it.

**Comment by bealdav** [comment]
I think @max3903 is right. 
That's the last module in the dependency order which has the last word.

In case of just manipulate the parent_id of the menu defined in agreement module

**Comment by ygol** [comment]
> 
> 
> I think @max3903 is right.
> That's the last module in the dependency order which has the last word.
> 
> In case of just manipulate the parent_id of the menu defined in agreement module

@max3903 @bealdav please check in the runbot http://runbot.odoo-community.org/runbot/build/3365913
no more agreement main menu

**Comment by bealdav** [comment]
Yes it's a feature. 

All the menus can't be in the main menu where there are a lot of menu

**Comment by max3903** [comment]
@ygol We can move the Agreement menu item back in the main menu in `agreement_legal_sale`. That's fine. Let's merge this one.

**Review by max3903 — APPROVED** [review]

**Pull request merged**